simulation/traffic_simulator/test/src/entity/test_entity_base.cpp

```diff
@@ -0,0 +1,768 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/entity/entity_base.hpp>
+#include <traffic_simulator/helper/helper.hpp>
+
+#include "../catalogs.hpp"
+#include "../entity_helper_functions.hpp"
+#include "../expect_eq_macros.hpp"
+#include "dummy_entity.hpp"
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+class EntityBaseTest : public testing::Test
+{
+protected:
+  EntityBaseTest()
+  : id(120659),
+    hdmap_utils(makeHdMapUtilsSharedPointer()),
+    pose(makeCanonicalizedLaneletPose(hdmap_utils, id)),
+    bbox(makeBoundingBox()),
+    status(makeCanonicalizedEntityStatus(hdmap_utils, pose, bbox)),
+    dummy("dummy_entity", status, hdmap_utils),
+    dummy_base(&dummy)
+  {
+  }
+
+  const lanelet::Id id;
+  std::shared_ptr<hdmap_utils::HdMapUtils> hdmap_utils;
+  traffic_simulator::lanelet_pose::CanonicalizedLaneletPose pose;
+  traffic_simulator_msgs::msg::BoundingBox bbox;
+  traffic_simulator::entity_status::CanonicalizedEntityStatus status;
+  DummyEntity dummy;
+  traffic_simulator::entity::EntityBase * dummy_base;
+};
+
+TEST_F(EntityBaseTest, appendDebugMarker)
+{
+  visualization_msgs::msg::MarkerArray markers{};
+
+  {
+    auto markers_copy = markers;
+    dummy.appendDebugMarker(markers);
+    EXPECT_EQ(markers.markers.size(), markers_copy.markers.size());
+  }
+
+  visualization_msgs::msg::Marker marker{};
+  marker.action = visualization_msgs::msg::Marker::ADD;
+  marker.type = visualization_msgs::msg::Marker::ARROW;
+  marker.pose.position.x = 10.0;
+  marker.pose.position.y = 10.0;
+
+  markers.markers.push_back(marker);
+
+  {
+    auto markers_copy = markers;
+    dummy.appendDebugMarker(markers);
+    EXPECT_EQ(markers.markers.size(), markers_copy.markers.size());
+  }
+}
+
+TEST_F(EntityBaseTest, asFieldOperatorApplication)
+{
+  EXPECT_THROW(dummy.asFieldOperatorApplication(), common::Error);
+}
+
+TEST_F(EntityBaseTest, get2DPolygon)
+{
+  const auto polygon = dummy.get2DPolygon();
+
+  std::vector<geometry_msgs::msg::Point> ref_poly{};
+  {
```

Comment:
Those braces seem unnecessary and do not add much to understanding the flow. Is there some other purpose for them?